File: angular-project1/src/app/Services/project.service.ts
// import { Injectable } from '@angular/core';

// @Injectable({
//   providedIn: 'root'
// })
// export class ProjectService {

//   constructor() { }
// }
import { Injectable } from '@angular/core';
import { HttpServerService } from '../http-server.service';

@Injectable({
  providedIn: 'root'
})
export class ProjectserviceService {

  constructor(private http:HttpServerService) { }
  // savethedb(serverName:string,projectName:string){
  //   return this.http.get('Project/readDB/' + serverName + '/' + projectName);
  // }
  projectName(code:string){
    debugger;
   return this.http.get('Project/projectName/' + code);
 }
 getTablesName(companycode:string,projectName:string){
   debugger;
  return this.http.get('Project/tablesName/' + companycode+ '/' + projectName );
  }
  getColum(projectCode:string,tableName:string){
    debugger;
    return this.http.get('Project/getColum/' +projectCode + '/' + tableName); 
  }
  // saveColum(projectCode:string,columToSave:string){
  //   debugger;
  //   return this.http.get('Project/saveColum/' +projectCode + '/' +columToSave);
  // }
//   saveSelectedFilter(projectCode:string,nameTableOfFilter:string,columNameListOfFilter:Array<string>){
//  debugger;
//   return this.http.post('Project/saveSelectedFilter/' +projectCode + '/' +nameTableOfFilter ,columNameListOfFilter);
// }
saveSelectedFilter(project:JSON){
  debugger;
 return this.http.post('Project/saveSelectedFilter' ,project);
 
}
 end(projectCode:string,columName:string,tableOfMailAddresss:string,emailColumn:string)
    {
return this.http.get('Project/end/'+projectCode+'/'+columName+'/'+tableOfMailAddresss+'/'+emailColumn);
    }
}

File: Servieces/ProjectService.cs
using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.SqlServer.Management.Smo;
using Newtonsoft.Json.Linq;
using Repositories;

namespace Servieces
{
    public class ProjectService : IProjectService
    {
        IProjectRepository projectRepository;
        public ProjectService(IProjectRepository projectRepository)
        {
            this.projectRepository = projectRepository;
        }

        public List<JObject> projectName(int code)
        {
            return projectRepository.projectName(code);
        }
        public Project tablesName(int code, string projectName)
        {
            return projectRepository.tablesName(code, projectName);
        }
        public List<string> getColum(int projectCode, string tableName)
        {
            return projectRepository.getColum(projectCode, tableName);
        }
        //public Boolean saveColum(int projectCode, string columName)
        //{
        //    return projectRepository.saveColum(projectCode, columName);
        //}
        public Boolean saveSelectedFilter(int projectCode, string tableName, List<string> columns)
        {
            return projectRepository.saveSelectedFilter(projectCode, tableName, columns);
        }
        public bool end(int projectCode, string columName, string tableOfMailAddress, string emailColumn)
        {
            return projectRepository.end(projectCode, columName, tableOfMailAddress, emailColumn);
        }
    }
}
